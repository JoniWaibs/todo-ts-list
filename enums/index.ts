export enum ItemState {
  DONE = 'done',
  TODO = 'todo',
  WIP = 'wip',
};