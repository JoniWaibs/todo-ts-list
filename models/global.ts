import { RenderResult } from '@testing-library/react';

export interface Global {
  appRenderWrapper(
    ui: JSX.Element,
    appProps?: any
  ): RenderResult,
}