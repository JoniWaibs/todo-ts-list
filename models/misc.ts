export interface StylableComponent {
  customClass?: string,
}