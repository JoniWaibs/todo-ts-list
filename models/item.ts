import { ItemState } from "@/enums/index";

export interface Item {
  id: string,
  name: string,
  state: ItemState,
}

export interface ListItem {
  items: Item[];
}