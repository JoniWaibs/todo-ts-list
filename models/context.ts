import { Dispatch, SetStateAction } from 'react';
import { Item } from './item';

export interface AppContextModel {
  items: Item[],
  setItems: Dispatch<SetStateAction<Item[]>>,
  doCompleteItem: (id: string) => void,
  deleteItem: (id: string) => void,
}

export interface AppProviderProps {
  children: JSX.Element | JSX.Element[]
}