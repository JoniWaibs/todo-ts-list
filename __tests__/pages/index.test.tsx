import { cleanup } from '@testing-library/react';
import Home from '@/pages/index'
import { Global } from '@/models/global'
import { Item as ItemProps } from '@/models/item';
import { ItemState } from '@/enums/index';

const { TODO, DONE, WIP } = ItemState;

declare const global: Global;

describe('App component', () => {
  let items: ItemProps[] = [];
  let setItems: () => void;
  let doCompleteItem: () => void;
  let deleteItem: () => void;

  beforeEach(() => {
    setItems = jest.fn();
    doCompleteItem = jest.fn();
    deleteItem = jest.fn();
    items = [{
      name: 'some little task',
      state: TODO,
      id: 'id-task-test-1',
    },
    {
      name: 'play football',
      state:  DONE,
      id: 'id-task-test-2',
    },
    {
      name: 'Learn Rust',
      state:  WIP,
      id: 'id-task-test-3',
    }]
  })

  afterEach(() => {
    cleanup();
  })

  test('Should render header with add tasks input', () => {
    const { container, getByText, getByLabelText } = global.appRenderWrapper(
      <Home />,
      {
        items,
        setItems,
        doCompleteItem,
        deleteItem,
      }
    );

    expect(container).toMatchSnapshot();
    getByText('TODO list')
    getByLabelText('Add your task')
  })

  test('Should render page with some tasks', () => {
    const { container, getByText } = global.appRenderWrapper(
      <Home />,
      {
        items,
        setItems,
        doCompleteItem,
        deleteItem,
      }
    );

    expect(container).toMatchSnapshot();
    items.forEach(({name}: ItemProps) => getByText(name))
  })
})