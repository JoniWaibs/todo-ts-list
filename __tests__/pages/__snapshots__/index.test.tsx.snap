// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render header with add tasks input 1`] = `
<div>
  <div>
    <h1>
      TODO list
    </h1>
    <input
      aria-label="Add your task"
      placeholder="Add your task"
      role="input"
      type="text"
      value=""
    />
    <button
      role="add-task-button"
    >
      Done
    </button>
  </div>
  <div>
    <div
      class="sc-bczRLJ kqGEgo"
    >
      <button
        role="complete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-check undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M5 12l5 5l10 -10"
          />
        </svg>
      </button>
      <p>
        some little task
      </p>
      <p />
      <button
        role="delete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-trash undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <line
            x1="4"
            x2="20"
            y1="7"
            y2="7"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
          <path
            d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
          />
          <path
            d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
          />
        </svg>
      </button>
    </div>
    <div
      class="sc-bczRLJ kqGEgo"
    >
      <button
        role="complete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-check undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M5 12l5 5l10 -10"
          />
        </svg>
      </button>
      <p>
        play football
      </p>
      <p>
        done
      </p>
      <button
        role="delete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-trash undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <line
            x1="4"
            x2="20"
            y1="7"
            y2="7"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
          <path
            d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
          />
          <path
            d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
          />
        </svg>
      </button>
    </div>
    <div
      class="sc-bczRLJ kqGEgo"
    >
      <button
        role="complete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-check undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M5 12l5 5l10 -10"
          />
        </svg>
      </button>
      <p>
        Learn Rust
      </p>
      <p />
      <button
        role="delete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-trash undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <line
            x1="4"
            x2="20"
            y1="7"
            y2="7"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
          <path
            d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
          />
          <path
            d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`App component Should render page with some tasks 1`] = `
<div>
  <div>
    <h1>
      TODO list
    </h1>
    <input
      aria-label="Add your task"
      placeholder="Add your task"
      role="input"
      type="text"
      value=""
    />
    <button
      role="add-task-button"
    >
      Done
    </button>
  </div>
  <div>
    <div
      class="sc-bczRLJ kqGEgo"
    >
      <button
        role="complete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-check undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M5 12l5 5l10 -10"
          />
        </svg>
      </button>
      <p>
        some little task
      </p>
      <p />
      <button
        role="delete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-trash undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <line
            x1="4"
            x2="20"
            y1="7"
            y2="7"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
          <path
            d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
          />
          <path
            d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
          />
        </svg>
      </button>
    </div>
    <div
      class="sc-bczRLJ kqGEgo"
    >
      <button
        role="complete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-check undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M5 12l5 5l10 -10"
          />
        </svg>
      </button>
      <p>
        play football
      </p>
      <p>
        done
      </p>
      <button
        role="delete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-trash undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <line
            x1="4"
            x2="20"
            y1="7"
            y2="7"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
          <path
            d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
          />
          <path
            d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
          />
        </svg>
      </button>
    </div>
    <div
      class="sc-bczRLJ kqGEgo"
    >
      <button
        role="complete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-check undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M5 12l5 5l10 -10"
          />
        </svg>
      </button>
      <p>
        Learn Rust
      </p>
      <p />
      <button
        role="delete-button"
      >
        <svg
          class="icon icon-tabler icon-tabler-trash undefined"
          fill="none"
          height="44"
          stroke="#2c3e50"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <line
            x1="4"
            x2="20"
            y1="7"
            y2="7"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
          <path
            d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
          />
          <path
            d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
